server/src/lib/config.rs

```diff
@@ -144,6 +148,77 @@ impl BlockDevice {
 
                 Ok((be, child))
             }
+            "crucible" => {
+                let targets: Vec<SocketAddrV4> = self
+                    .options
+                    .get("targets")
+                    .ok_or_else(|| {
+                        ParseError::KeyNotFound(
+                            "targets".to_string(),
+                            "options".to_string(),
+                        )
+                    })?
+                    .as_array()
+                    .ok_or_else(|| {
+                        ParseError::AsError(
+                            "targets".to_string(),
+                            "as_array".to_string(),
+                        )
+                    })?
+                    .to_vec()
+                    .iter()
+                    .map(|x| {
+                        x.as_str()
+                            .ok_or_else(|| {
+                                ParseError::AsError(
+                                    "x".to_string(),
+                                    "as_str".to_string(),
+                                )
+                            })?
+                            .to_string()
+                            .parse()
+                            .map_err(|e| {
+                                ParseError::AsError(
+                                    "x".to_string(),
+                                    format!("parse(): {:?}", e),
+                                )
+                            })
+                    })
+                    .collect::<Result<Vec<SocketAddrV4>, ParseError>>()?;
```

Comment:
Not a problem for right now, but boy-howdy is this verbose.  Let's put our heads together on how we might improve the ergonomics for config stuff down the line, since these cases will only become more complex.